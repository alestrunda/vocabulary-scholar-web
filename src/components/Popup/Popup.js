/*global window*/

import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Cover from '../Cover';
import Paper from 'material-ui/Paper';

const Popup = props => {
  const handleCoverClick = () => {
    props.onClose && props.onClose();
  };

  const popupTop = window.pageYOffset;
  const popupTopMargin = 15;

  return (
    <React.Fragment>
      <Cover isActive={props.isActive} onClick={handleCoverClick} />
      <Paper
        zDepth={2}
        style={{
          top: popupTop,
          borderRadius: 5,
          maxHeight: `calc(100% - ${popupTopMargin * 2}px - ${popupTop}px) `,
          ...props.style,
        }}
        className={classNames('popup', props.className, {
          active: props.isActive,
        })}
      >
        <h2 className="mb25 text-center">{props.title}</h2>
        <div className="popup__content">{props.children}</div>
      </Paper>
    </React.Fragment>
  );
};

Popup.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  className: PropTypes.string,
  isActive: PropTypes.bool,
  title: PropTypes.string,
  onClose: PropTypes.func,
  style: PropTypes.object,
};

Popup.defaultProps = {
  isActive: false,
};

export default Popup;
