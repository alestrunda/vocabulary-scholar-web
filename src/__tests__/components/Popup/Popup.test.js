import React from 'react';
import { shallow } from 'enzyme';
import Popup from '../../../components/Popup';

it('renders correctly', () => {
  const popup = shallow(
    <Popup>
      <p>Lorem impsum</p>
    </Popup>
  );
  expect(popup).toMatchSnapshot();
});

it('closes when clicked outside', () => {
  const onClose = jest.fn();
  const popup = shallow(
    <Popup onClose={onClose}>
      <p>Lorem impsum</p>
    </Popup>
  );
  popup.find('Cover').simulate('click');
  expect(onClose).toHaveBeenCalled();
});

it('has active class', () => {
  const popup = shallow(
    <Popup isActive>
      <p>Lorem impsum</p>
    </Popup>
  );
  expect(popup.find('Paper').hasClass('active')).toBe(true);
});
