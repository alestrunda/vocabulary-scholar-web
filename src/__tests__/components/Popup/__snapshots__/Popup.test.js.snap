// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Popup
    isActive={false}
  >
    <p>
      Lorem impsum
    </p>
  </Popup>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Cover
          isActive={false}
          onClick={[Function]}
          rate={0.5}
        />,
        <Paper
          circle={false}
          className="popup"
          rounded={true}
          style={
            Object {
              "borderRadius": 5,
              "maxHeight": "calc(100% - 30px - 0px) ",
              "top": 0,
            }
          }
          transitionEnabled={true}
          zDepth={2}
        >
          <h2
            className="mb25 text-center"
          />
          <div
            className="popup__content"
          >
            <p>
              Lorem impsum
            </p>
          </div>
        </Paper>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isActive": false,
          "onClick": [Function],
          "rate": 0.5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h2
              className="mb25 text-center"
            />,
            <div
              className="popup__content"
            >
              <p>
                Lorem impsum
              </p>
            </div>,
          ],
          "circle": false,
          "className": "popup",
          "rounded": true,
          "style": Object {
            "borderRadius": 5,
            "maxHeight": "calc(100% - 30px - 0px) ",
            "top": 0,
          },
          "transitionEnabled": true,
          "zDepth": 2,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "mb25 text-center",
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Lorem impsum
              </p>,
              "className": "popup__content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lorem impsum",
              },
              "ref": null,
              "rendered": "Lorem impsum",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Cover
            isActive={false}
            onClick={[Function]}
            rate={0.5}
          />,
          <Paper
            circle={false}
            className="popup"
            rounded={true}
            style={
              Object {
                "borderRadius": 5,
                "maxHeight": "calc(100% - 30px - 0px) ",
                "top": 0,
              }
            }
            transitionEnabled={true}
            zDepth={2}
          >
            <h2
              className="mb25 text-center"
            />
            <div
              className="popup__content"
            >
              <p>
                Lorem impsum
              </p>
            </div>
          </Paper>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isActive": false,
            "onClick": [Function],
            "rate": 0.5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h2
                className="mb25 text-center"
              />,
              <div
                className="popup__content"
              >
                <p>
                  Lorem impsum
                </p>
              </div>,
            ],
            "circle": false,
            "className": "popup",
            "rounded": true,
            "style": Object {
              "borderRadius": 5,
              "maxHeight": "calc(100% - 30px - 0px) ",
              "top": 0,
            },
            "transitionEnabled": true,
            "zDepth": 2,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "mb25 text-center",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Lorem impsum
                </p>,
                "className": "popup__content",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lorem impsum",
                },
                "ref": null,
                "rendered": "Lorem impsum",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
